internal/scheduler/queue_scheduler.go

```diff
@@ -210,7 +210,10 @@ func (it *QueuedGangIterator) Peek() (*schedulercontext.GangSchedulingContext, e
 
 		// Skip this job if it's known to be unschedulable.
 		if len(it.schedulingContext.UnfeasibleSchedulingKeys) > 0 {
-			schedulingKey := it.schedulingContext.SchedulingKeyFromLegacySchedulerJob(job)
+			schedulingKey, ok := job.GetSchedulingKey()
+			if !ok {
+				schedulingKey = it.schedulingContext.SchedulingKeyFromLegacySchedulerJob(job)
```

Comment:
This fallback is necessary to handle api.Job, which doesn't have a pre-computed schedulingKey associated with it.